akka-docs/src/main/paradox/scala/general/actor-systems.md

```diff
@@ -107,14 +107,153 @@ which is a single point of contention if over-used.
+ * Orange - Waiting state
+ * Green - Runnable state
+
+The thread information was recorded using the YourKit profiler, however any good JVM profiler 
+has this feature (including the free and bundled with the Oracle JDK VisualVM, as well as Oracle Flight Recorder). 
+
+The orange portion of the thread shows that it is idle. Idle threads are fine -
+they're ready to accept new work. However, large amounts of Turquoise (sleeping) threads are very bad!
+
+![dispatcher-behaviour-on-good-code.png](../../images/dispatcher-behaviour-on-bad-code.png)
+
+The app is exposed to the load of `i: Int` messages which will block these threads. 
+For example "`default-akka.default-dispatcher2,3,4`"
+are going into the blocking state, after having been idle. It can be observed
+that the number of new threads increases, "`default-akka.actor.default-dispatcher 18,19,20,...`" 
+however they go to sleep state immediately, thus wasting the
+resources.
+
+The number of such new threads depends on the default dispatcher configuration,
+but it will likely not exceed 50. Since many requests are being processed, the entire
+thread pool is starved. The blocking operations dominate threads in the dispatcher 
+such that it has no thread available to handle other requests.
+
+In essence, the `Thread.sleep` operation has dominated all threads and caused anything 
+executing on the default dispatcher to starve for resources (including any actor
+that you have not configured an explicit dispatcher for).
+
+## Solution: Dedicated dispatcher for blocking operations
+
+In `application.conf`, the dispatcher dedicated to blocking behaviour should
+be configured as follows:
+
+```conf
+my-blocking-dispatcher {
+  type = Dispatcher
+  executor = "thread-pool-executor"
+  thread-pool-executor {
+    // or in Akka 2.4.2+
+    fixed-pool-size = 16
+  }
+  throughput = 100
+}
+```
+
+There are many dispatcher options available which can be found in @ref[Dispatchers](../dispatchers.md).
+
+Here `thread-pool-executor` is used, which has a hardcoded limit of threads. It keeps a set number of threads
+available that allow for safe isolation of the blocking operations. The size settings should depend on the app's
+functionality and the number of cores the server has.
+
+Whenever blocking has to be done, use the above configured dispatcher
+instead of the default one:
+
+@@snip [BlockingDispatcherSample.scala](../../../../test/scala/docs/actor/BlockingDispatcherSample.scala) { #separate-dispatcher }
+
+The thread pool behaviour is shown in the figure.
+
+![dispatcher-behaviour-on-good-code.png](../../images/dispatcher-behaviour-on-good-code.png)
+
+the normal requests are easily handled by the default dispatcher - the
+green lines, which represent the actual execution.
+
+When blocking operations are issued, the `my-blocking-dispatcher`
+starts up to the number of configured threads. It handles sleeping. After
+a certain period of nothing happening to the threads, it shuts them down.
```

Comment:
When blocking operations are run on the `my-blocking-dispatcher`,
it uses the threads (up to the configured limit) to handle these operations.
The sleeping in this case is nicely isolated to just this dispatcher, and the default one remains unaffected,
allowing the rest of the application to proceed as if nothing bad was happening. After
a certain period idleness, threads started by this dispatcher will be shut down.